include/core/CCompressedLfuCache.h

```diff
@@ -149,12 +149,19 @@ class CCompressedLfuCache {
 
             // Update the item counts using the Space-Saving algorithm.
             std::size_t count{1};
-            if (this->full(itemMemoryUsage)) {
+            std::size_t maxEvictedCount{0};
+            while (this->full(itemMemoryUsage)) {
```

Comment:
In fact, maybe with the while loop approach we simply change `full` to not have a margin at all. The margin was to allow for variation in item size.

Reply:
Yes, I like the idea of not having a margin.  I'll make that change.

> The margin was to allow for variation in item size.

There was another nasty edge case before which was if items gradually increase in size over time.  Since we'd only evict one item per item added we could eventually exceed the limit by quite a lot, because each single item added would be freeing up a smaller amount of memory.  I know this scenario isn't likely for inference responses, but it might have happened for some future use of the class.

Reply:
I guess this goes back to the assumption that the cache really only holds items << cache size. I was thinking size < 0.01 * cache size would be very conservative. Really if we did want to try and handle other cases better I would probably advocate being more clever about item size when it comes to deciding what to cache. I don't think we should ever be in a case where this assumption is violated for inference, but it is worth bearing in mind.